core/src/main/java/org/infinispan/remoting/inboundhandler/GlobalInboundInvocationHandler.java

```diff
@@ -135,7 +135,7 @@ private void handleReplicableCommand(Address origin, ReplicableCommand command,
          //we must/can run in this thread
          runnable.run();
       } else {
-         remoteCommandsExecutor.execute(runnable);
+         blockingExecutor.execute(runnable);
```

Comment:
So we assume any `ReplicableCommand` is blocking?

Reply:
Why do you say that, sorry? It should only do this if `canBlock` was true.

Reply:
Sorry, I was only looking at the diff and I forgot that `ReplicableCommand` has `canBlock()`, I thought it was in `CacheRpcCommand`.